src/js/components/List/List.js

```diff
@@ -148,6 +148,7 @@ const List = React.forwardRef(
       onKeyDown,
       onMore,
       onOrder,
+      hideIndex = false,
```

Comment:
@grommet/devs any thoughts on naming for this prop?

Reply:
showIndex,
displayIndex
displayIndexNumber
isVisable 


just some ideas we can think more but I like this functionality

Reply:
I like `showIndex` and then just have it default to true. There are other props in grommet where we use the word `show` so this might help with consistency in the language we use.

Reply:
sounds good ☝️